### PR TITLE
fix bug white page when selecting SJB | refactoring code

### PR DESCRIPTION
**Pull request by bendaamerahm**
- fix bug white page when selecting SJB

- refactoring code

### CODE REVIEW

File: src/components/dialogs/input-hooks.js [hunk 1]
```diff
@@ -282,6 +284,7 @@ export const useConnectivityValue = ({
     }, [inputForm.toggleClear]);
 
     useEffect(() => {
+        dispatch(setFirstBuildNode());
```
Comment (sBouzols) [inline src/components/dialogs/input-hooks.js:287]: Why do we need this ?

File: src/components/dialogs/input-hooks.js [hunk 2]
```diff
@@ -273,6 +273,8 @@ export const useConnectivityValue = ({
     const [errorBusBarSection, setErrorBusBarSection] = useState();
     const intl = useIntl();
     const studyUuid = useSelector((state) => state.studyUuid);
+    const currentNode = useSelector((state) => state.currentTreeNode);
```
Comment (sBouzols) [inline src/components/dialogs/input-hooks.js:276]: I would be instinctively against using Redux Store inside our custom hooks. But maybe it's ok ?
We could keep passing Uuids as hooks props no ? Seems ok for me.

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/dialogs/input-hooks.js (hunk 1 above)
Comment (bendaamerahm) [reply src/components/dialogs/input-hooks.js:287]: code not updated , ill do the cleaning and refactor

File: src/redux/reducer.js [hunk 3]
```diff
@@ -154,6 +156,9 @@ export const reducer = createReducer(initialState, {
         state.networkModificationTreeModel =
             action.networkModificationTreeModel;
         state.networkModificationTreeModel.setBuildingStatus();
+        const firstBuiltNode = getFirstBuildNode(state);
+        state.currentTreeNode = { ...firstBuiltNode };
+        state.firstBuiltNode = { ...firstBuiltNode };
```
[the remark this answers is not in the document]
Comment (bendaamerahm) [reply src/redux/reducer.js:161]: code not updated im waiting the other PR to pass, ill do the cleaning and refactor

File: src/redux/reducer.js [hunk 4]
```diff
@@ -346,14 +352,34 @@ export const reducer = createReducer(initialState, {
     [OPEN_NETWORK_AREA_DIAGRAM]: (state, action) => {
         state.voltageLevelsIdsForNad = action.voltageLevelsIdsForNad;
     },
+    [FIRST_BUILT_NODE]: (state) => {
+        if (state.networkModificationTreeModel) {
+            state.firstBuiltNode = getFirstBuildNode(state);
```
[the remark this answers is not in the document]
Comment (bendaamerahm) [reply src/redux/reducer.js:357]: code not updated im waiting the other PR to pass, ill do the cleaning and refactor

File: src/redux/reducer.js [hunk 5]
```diff
@@ -167,6 +172,7 @@ export const reducer = createReducer(initialState, {
             );
             newModel.updateLayout();
             state.networkModificationTreeModel = newModel;
+            synchCurrentTreeNode(state);
```
[the remark this answers is not in the document]
Comment (bendaamerahm) [reply src/redux/reducer.js:175]: my bad

**Review by sBouzols — APPROVED** [review]

**Pull request merged**